src/prefect/events/clients.py

```diff
@@ -346,6 +346,47 @@ async def _emit(self, event: Event) -> None:
                     await asyncio.sleep(1)
 
 
+class AssertingPrefectEventsClient(PrefectEventsClient):
```

Comment:
I think a name like this communicates the functionality a little clearer.
```suggestion
class AssertingPassthroughEventsClient(PrefectEventsClient):
```